### ISSUE TITLE
Unable to load backup

### ISSUE THREAD

**Issue by GlitzaFlitzer** [jellyfin/jellyfin-plugin-playbackreporting#84]
Hey,
today I created a backup of my playback reporting with the aim of deleting reports that were not desirable in it. After editing, I deleted all my data and wanted to load the tsv file again using the "Load Backup Data" tab. Unfortunately, nothing happens. Is this option possibly buggy or is it just me? Thanks! 

I use version 14.0

**Comment by ibndiaye** [comment]
Yep, I'm also experiencing this issue on jellyfin 10.9.4, I'll be able to post some more info soon.

**Comment by ibndiaye** [comment]
> Yep, I'm also experiencing this issue on jellyfin 10.9.4, I'll be able to post some more info soon.

Jellyfin server running on ubuntu lxc:
```
Server version
10.9.4
Web version
10.9.4
Build version
10.9.4

```
Logs when loading the backup:
```
[2024-06-05 16:55:07.544 +00:00] [INF] PlaybackReportingActivityController Loaded
[2024-06-05 16:55:07.547 +00:00] [INF] Default journal_mode for "/var/lib/jellyfin/data/playback_reporting.db" is "delete"
[2024-06-05 16:55:10.106 +00:00] [INF] PlaybackReportingActivityController Loaded
[2024-06-05 16:55:10.109 +00:00] [INF] Default journal_mode for "/var/lib/jellyfin/data/playback_reporting.db" is "delete"
[2024-06-05 16:55:18.338 +00:00] [INF] PlaybackReportingActivityController Loaded
```

**Comment by Souperwilli** [comment]
I have the same issue running Jellyfin 10.10.13 in a Docker container, getting the same logs as @ibndiaye.
Are there any news on this topic?

**Comment by leolifely** [comment]
I have this issue as well, with Jellyfin 10.10.3 on Ubuntu Server 24.04.1 LTS, with these logs: 
`Dec 24 17:52:43 leoli jellyfin[52303]: [17:52:43] [INF] Default journal_mode for /var/lib/jellyfin/data/playback_reporting.db is delete
Dec 24 17:52:43 leoli jellyfin[52303]: [17:52:43] [INF] PlaybackReportingActivityController Loaded
Dec 24 17:52:43 leoli jellyfin[52303]: [17:52:43] [INF] Default journal_mode for /var/lib/jellyfin/data/playback_reporting.db is delete
Dec 24 17:52:46 leoli jellyfin[52303]: [17:52:46] [INF] PlaybackReportingActivityController Loaded
Dec 24 17:52:46 leoli jellyfin[52303]: [17:52:46] [INF] Default journal_mode for /var/lib/jellyfin/data/playback_reporting.db is delete`

**Comment by Flo56958** [comment]
Same journal_mode issue when trying to load a backup with plugin version 15.0.0.0 and Jellyfin 10.10.4.

For anyone coming across this issue as well, I have created a workaround by manually editing playback_reporting.db with this short python script: 
```python
#EDIT THIS
DATABASE = "playback_reporting.db"
BACKUP = "backup.tsv"

import sqlite3
import csv
con = sqlite3.connect(DATABASE)
cursor = con.cursor()

with open(BACKUP) as fd:
    rd = csv.reader(fd, delimiter="\t", quotechar='"')
    for row in rd:
        query = f"INSERT INTO PlaybackActivity VALUES ({", ".join([f'"{x.replace('"', '""')}"' for x in row])});"
        cursor.execute(query)

con.commit()
```
It worked for me while Jellyfin was shut down. Hope it helps.

**Comment by nazgul6** [comment]
@Flo56958 - thanks! this worked to import the backup containing history for me as well. However, the users from the backup were imported as new users with names 'Not Known' instead of the history simply appended to existing users, which is what I would have expected given same user names in the backup file and the newly setup jellyfin instance. Did you run into this as well? Thanks!

**Comment by nazgul6** [comment]
@Flo56958 , perfect, thanks, sorted it.

For future reference, to list the UUIDS from jellyfin.db, the command is:

`sqlite3 jellyfin.db "SELECT Id, Username FROM Users;"`

The UUIDs will be listed with dashes and capital letters, one should remove the dashes (I also made them all lower case to match the backup export).

Tip:
When editing .tsv from playback reporting to run find+replace, this is best done through a text editor as opposed to excel. Excel will mess up the first column of data, which is DD/MM/YYYY hh:mm:ss with seconds followed by seven further digits. If those are truncated (Excel seems to cut them off no matter what), the plugin won't pick up any of the data when re-imported.

**Comment by blazklemencic** [comment]
while the python script did work fine, I believe that this backup import process is a bit unpleasant to say the least.
